### DISCUSSION TITLE
Why use GPS data as a constraint term rather than treating it as the ground truth or as initial values with specific offset constraints?

### DISCUSSION THREAD

**Discussion by StonerLing** [openMVG/openMVG#2377] [General]
I notice that there is a cost function `PoseCenterConstraintCostFunction` that can utilize GPS positions as constraint terms. However, judging from the implementation of this function, we want the optimized camera positions to be as close as possible to the GPS positions. This also means that we believe the GPS coordinates have a sufficiently reliable accuracy. So, why not consider the following treatments: 

- Use the GPS positions as the ground truth and exclude them from optimization; 
- Use the GPS positions as initial values, and we might know their errors in advance (for example, 10 cm), so we can set bounds on the optimized camera positions. This can be achieved through Ceres `setParameterLowerBound` and `setParameterUpperBound`, although it will significantly increase the computation time.

**Comment by pmoulon** [comment]
Thank you for your feedback.
The initial implementation has been done a first solution, the solution that you are proposing seems indeed another solution we can use (depending of the quality of the GPS value).

Feel free to make a proposal and sent a PR to us so we can test, review and implement such a constraint.
